src/components/icon-tile-logo/index.tsx

```diff
@@ -0,0 +1,31 @@
+import ProductIcon from 'components/product-icon'
+import { ProductSlug } from 'types/products'
+import IconTile from 'components/icon-tile'
+
+type ProductWithIconSlug = Exclude<ProductSlug, 'sentinel'>
+interface IconTileLogoProps {
+  product: ProductWithIconSlug
+}
+
+/**
+ * Use the IconTileLogo component when representing the product in our interfaces.
+ *
+ * Renders a product icon in a brand-colored tile. Note that this IconTileLogo
+ * is currently styled to match dev-portal designs, and differs in size and
+ * background coloration from the IconTileLogo documented in the
+ * HashiCorp Design System.
+ *
+ * ref: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components-%5BWIP%5D?node-id=1377%3A11992
```

Comment:
Added notes on the `IconTile` & `IconTileLogo` implementations, to acknowledge they're not 1:1 with the HashiCorp Design System Figma yet.